### ISSUE TITLE
Single request to obtain a snapshot ?

### ISSUE THREAD

**Issue by nodecentral** [steets250/vera-ring#1]
Hi,

Thanks so much for sharing the plug-in, I was just wondering if you had any standalone Lua code option, that could go off and obtain a snapshot of what all registered cameras can see ? 

This would mirror some foscam code I already have on Vera that uses their api to obtain a snapshot and then write it to a local drive, after which I then retrieve it and add it into an email to send out.

**Comment by steets250** [comment]
Hi,

Thanks for your interest in the plug-in. Unfortunately, the only functionality I was able to implement was a super basic light / doorbell status indicator. I used [dgrief's ring plugin](https://github.com/dgreif/ring) as a base when trying to get an API working in Lua, so I think you could find the right calls necessary in that repo to make that functionality happen.

**Comment by nodecentral** [comment]
Thanks for responding 

I’ve not installed it yet on my Vera , but what exactly do you mean by it works as a ‘ super basic light / doorbell status indicator.’? 

Do you mean it responds to a door bell push and camera motion ? ( like a security/motion sensor ?)

**Comment by steets250** [comment]
To be quite honest, I don't even think I got the doorbell status working. I originally wanted to create the program to allow me to automate turning on and off the Ring notifications based on if the mode was Home/Away. I struggled trying to figure out Lua and the Vera documentation, so I didn't really get that far. So all the plugin does is allows you to turn on/off a Ring light.

**Comment by nodecentral** [comment]
Thanks again 

If it helps, I have some experience with Vera, so if you can get the lua part to obtain a JSON response or some-other type of related response,  then it might be possible to build some basic functionality?  There is a core group on the Vera forum that seem pretty knowledgeable and responsive..

**Comment by steets250** [comment]
Yeah that would be great. I'll try and port a basic version of the Ring API over, and then see if I can get some help with the Vera aspect.

**Comment by nodecentral** [comment]
Just to let you know there is a ring plugin thread on the Vera forum - https://community.getvera.com/t/ring-video-doorbell-vera/188907/106 - where lower down I posted a link to you plugin. With over 2.3k views I would say you have a good audience of willing testers.. So you might want to create a entry in the Plugin Development support section - > https://community.getvera.com/c/plugins-and-plugin-development/129

**Comment by steets250** [comment]
It's great to see that there's interest for this. Sorry to disappoint everyone that it's not more fleshed out 😪

I'm not sure that I'll have the time to make this plugin, but I can definitely give the Lua API a try and see if someone else can take it from there.

**Comment by nodecentral** [comment]
That’ll be great,

If you can find a way to get the information (ideally via a push and a pull ) then I certainly think someone could then help or take it over.

I’ll certainly see what I can do - even if initially it’s just used as a module that can be called via the Vera start up - it’s a good start :)

On 10 Dec 2020, at 08:51, Steven Steiner <notifications@github.com> wrote:

﻿

It's great to see that there's interest for this. Sorry to disappoint everyone that it's not more fleshed out 😪

I'm not sure that I'll have the time to make this plugin, but I can definitely give the Lua API a try and see if someone else can take it from there.
